Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Quality/QualityCVarGroup.h>
+#include <AzFramework/Quality/QualitySystemBus.h>
+
+#include <AzCore/Settings/SettingsRegistryImpl.h>
+#include <AzCore/Settings/SettingsRegistryVisitorUtils.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzCore/Console/IConsole.h>
+#include <AzCore/StringFunc/StringFunc.h>
+
+namespace AzFramework
+{
+    QualityCVarGroup::QualityCVarGroup(AZ::SettingsRegistryInterface* registry, AZ::IConsole* console, AZStd::string_view name, AZStd::string_view path)
+        : m_console(console)
+        , m_numQualityLevels(0)
+        , m_path(path)
+        , m_qualityLevel(0)
+        , m_settingsRegistry(registry)
+    {
+        AZ_Assert(registry, "QualityCVarGroup requireds a valid settings registry.");
+        AZ_Assert(console, "QualityCVarGroup requireds a valid settings registry.");
+
+        using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+
+        // optional default quality level
+        registry->Get(m_qualityLevel, FixedValueString::format("%.*s/Default", AZ_STRING_ARG(path)));

```

Comment:
Here is the part that can be changed to use `GetObject` if m_qualityLevel is made an enum.
```suggestion
        // optional default quality level
        registry->GetObject(m_qualityLevel, FixedValueString::format("%.*s/Default", AZ_STRING_ARG(path)));

```